src/main/java/io/gravitee/repository/redis/common/RedisConnectionFactory.java

```diff
@@ -36,29 +46,46 @@
 
     private final String propertyPrefix;
 
+    private final static String SENTINEL_HOST_SEPARATOR = ",";
+
     public RedisConnectionFactory(String propertyPrefix) {
         this.propertyPrefix = propertyPrefix + ".redis.";
     }
 
     @Override
     public org.springframework.data.redis.connection.RedisConnectionFactory getObject() throws Exception {
-        JedisConnectionFactory jedisConnectionFactory = new JedisConnectionFactory();
-        jedisConnectionFactory.setHostName(readPropertyValue(propertyPrefix + "host", String.class, "localhost"));
-        jedisConnectionFactory.setPort(readPropertyValue(propertyPrefix + "port", int.class, 6379));
-        jedisConnectionFactory.setPassword(readPropertyValue(propertyPrefix + "password", String.class, null));
-        jedisConnectionFactory.setTimeout(readPropertyValue(propertyPrefix + "timeout", int.class, -1));
-        jedisConnectionFactory.setUseSsl(readPropertyValue(propertyPrefix + "ssl", boolean.class, false));
-
-        int poolMax = readPropertyValue(propertyPrefix + "pool.max", int.class, 256);
-
-        JedisPoolConfig poolConfig = new JedisPoolConfig();
-        poolConfig.setMaxTotal(poolMax);
-        poolConfig.setBlockWhenExhausted(false);
-        jedisConnectionFactory.setPoolConfig(poolConfig);
-
-        jedisConnectionFactory.afterPropertiesSet();
-
-        return jedisConnectionFactory;
+        final LettuceConnectionFactory lettuceConnectionFactory;
+
+        final String sentinelNodesProperty = readPropertyValue(propertyPrefix + "sentinel.nodes", String.class, "");
+        if (StringUtils.isBlank(sentinelNodesProperty)) {
+            // Standalone Redis
+            RedisStandaloneConfiguration standaloneConfiguration = new RedisStandaloneConfiguration();
+            standaloneConfiguration.setHostName(readPropertyValue(propertyPrefix + "host", String.class, "localhost"));
+            standaloneConfiguration.setPort(readPropertyValue(propertyPrefix + "port", int.class, 6379));
+            standaloneConfiguration.setPassword(readPropertyValue(propertyPrefix + "password", String.class, null));
+
+            lettuceConnectionFactory = new LettuceConnectionFactory(standaloneConfiguration, buildLettuceClientConfiguration());
+        } else {
+            // Sentinels + Redis master / replicas
+            List<String> sentinelNodes = Arrays.asList(sentinelNodesProperty.split(SENTINEL_HOST_SEPARATOR));
+            String redisMaster = readPropertyValue(propertyPrefix + "sentinel.master", String.class, "mymaster");
```

Comment:
Is `mycluster` really the default name for sentinel ?

Reply:
No, a default value does not make sense for the master name :).  
This is the name used in all the examples in the Redis documentation.

I can default it to blank and clearly indicate that this parameter is mandatory. Perhaps throw an exception if missing ?

Reply:
That would make more sense yes, thanks.